sdk/core/azure-core/azure/core/pipeline/policies/_universal.py

```diff
@@ -267,6 +268,105 @@ def on_response(self, request, response):
                 _LOGGER.debug("Failed to log response: %s", repr(err))
 
 
+class HttpLoggingPolicy(SansIOHTTPPolicy):
+    """The Pipeline policy that handles logging of HTTP requests and responses.
+    """
+
+    DEFAULT_HEADERS_WHITELIST = set([
+        "x-ms-client-request-id",
+        "x-ms-return-client-request-id",
+        "traceparent",
+        "Accept",
+        "Cache-Control",
+        "Connection",
+        "Content-Length",
+        "Content-Type",
+        "Date",
+        "ETag",
+        "Expires",
+        "If-Match",
+        "If-Modified-Since",
+        "If-None-Match",
+        "If-Unmodified-Since",
+        "Last-Modified",
+        "Pragma",
+        "Request-Id",
+        "Retry-After",
+        "Server",
+        "Transfer-Encoding",
+        "User-Agent"
+    ])
+    REDACTED_PLACEHOLDER = "REDACTED"
+
+    def __init__(self, logger=None, **kwargs):  # pylint: disable=unused-argument
+        self.logger = logger or logging.getLogger(
+            "azure.core.pipeline.policies.http_logging_policy"
+        )
+        self.allowed_query_params = set()
+        self.allowed_header_namers = set(HttpLoggingPolicy.DEFAULT_HEADERS_WHITELIST)
+
+    def _redact_query_param(self, key, value):
+        lower_case_allowed_query_params = [
+            param.lower() for param in self.allowed_query_params
+        ]
+        return value if key.lower() in lower_case_allowed_query_params else HttpLoggingPolicy.REDACTED_PLACEHOLDER
+
+    def _redact_header(self, key, value):
+        lower_case_allowed_header_namers = [
+            header.lower() for header in self.allowed_header_namers
+        ]
```

Comment:
We should do this in init so we only need to run it once?

Reply:
The idea is that you are able to add/remove more headers after creation, then you can't compute this list in the `__init__`
```python
policy = HttpLoggingPolicy()
policy.allowed_header_namers.add("x-ms-something")
```

Reply:
Please put this comment on https://github.com/Azure/azure-sdk-for-python/pull/8027 if you still think it's necessary. Thanks!